### ISSUE TITLE
PF4 Select: Let set an Icon for Toggle

### ISSUE THREAD

**Issue by aljesusg** [patternfly/patternfly-react#1767]
Feature to let personalize icon to select component.

**Comment by rachael-phillips** [comment]
Hi @aljesusg do you mind telling us more about your use case and need here? We can't see the example in the PR - would you mind updating that?

**Comment by aljesusg** [comment]
Set another Icon not the caret, for example Angle Down, we want to have the same icons for dropdowns

**Comment by LHinson** [comment]
hey @aljesusg! Trying to understand a bit more here. Is the intent to keep your previous icons an artifact of incremental migration? i.e. you are using PF4 components alongside PF3 and need to address inconsistencies between the two? Thanks in advance for the extra info!

**Comment by aljesusg** [comment]
Yes @LHinson , it is to try to have a correct user experience while making the migration to PF4.
However, if PF4 has this icon as a core-style we can close this issue.

**Comment by tlabaj** [comment]
@aljesusg are you looking to change the default Icon?

**Comment by dlabaj** [comment]
@LHinson Any updates on this?

**Comment by aljesusg** [comment]
@dlabaj I closed this because I didn't know that the PF4 guideline indicates that this is the icon for Select, but I saw that @rachael-phillips added this to the the Sprint If you think that this feature is ok for PF4 I can do it

**Comment by boaz0** [comment]
I will give this +1 as this is what we need for the PF4 filter we are using.
I tried to solve this using `InputGroup` and `InputGroupText` but it doesn't look perfect as having the icon inside the select toggle.

If no one is working on this, can you assign this to me.

related to https://github.com/project-koku/koku-ui/issues/946

//cc @dlabrecq

**Comment by dlabrecq** [comment]
Attn: @LHinson @rachael-phillips 

Cost Management needs the ability to add a filter icon in order to implement the PatternFly filter design shown below.

https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/filters#attribute-value-textbox-filters
![Screen Shot 2019-08-14 at 10 19 59 AM](https://user-images.githubusercontent.com/17481322/63028615-273cdb80-be7d-11e9-9694-4288cd3d5a74.png)

**Comment by rachael-phillips** [comment]
HI @dlabrecq @aljesusg @boaz0 we have opened https://github.com/patternfly/patternfly-next/issues/2152 on the Core side. We'd like to address this first before addressing 1767. We'll add support for the icon in core by creating a new element in the select component for it, and that element will apply style (spacing) to the icon. Then #1767 can be updated to pull in that change and make the icon a thing a user can pass to the select component.

cc @mcoker

**Comment by dlabrecq** [comment]
@mcoker Here are a couple screen shots of what Boaz created in an attempt to match the PatternFly design. The icons are clearly outside the dropdown and input text.

<img width="560" alt="Screen Shot 2019-08-16 at 11 08 51 AM" src="https://user-images.githubusercontent.com/17481322/63177868-d06b0980-c016-11e9-98cd-6b891a491a5f.png">

This is another example of a form input field with a percentage. Some of our inputs use dollar symbols, instead.
